src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,305 @@
+     */
+    @PutMapping("/courses/{courseId}/tutorial-groups")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<TutorialGroup> update(@PathVariable Long courseId, @RequestBody @Valid TutorialGroup tutorialGroup) {
+        log.debug("REST request to update TutorialGroup : {}", tutorialGroup);
+        if (tutorialGroup.getId() == null) {
+            throw new BadRequestException("A tutorial group cannot be updated without an id");
+        }
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdWithTeachingAssistantAndRegistrationsElseThrow(tutorialGroup.getId());
+        checkTutorialCourseIdMatchesPathId(courseId, tutorialGroupFromDatabase);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+
+        trimStringFields(tutorialGroup);
+        if (!tutorialGroupFromDatabase.getTitle().equals(tutorialGroup.getTitle())) {
+            checkTitleIsUnique(tutorialGroup);
+        }
+        overrideValues(tutorialGroup, tutorialGroupFromDatabase);
+
+        var updatedTutorialGroup = tutorialGroupRepository.save(tutorialGroupFromDatabase);
+        return ResponseEntity.ok(updatedTutorialGroup);
+    }
+
+    /**
+     * DELETE courses/:courseId/tutorial-groups/:tutorialGroupId/deregister/:studentLogin : deregister a student from a tutorial group.
+     *
+     * @param courseId        the id of the course to which the tutorial groups belong to
+     * @param tutorialGroupId the id of the tutorial group
+     * @param studentLogin    the login of the student to deregister
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @DeleteMapping("/courses/{courseId}/tutorial-groups/{tutorialGroupId}/deregister/{studentLogin:" + Constants.LOGIN_REGEX + "}")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<Void> deregisterStudent(@PathVariable Long courseId, @PathVariable Long tutorialGroupId, @PathVariable String studentLogin) {
+        log.debug("REST request to deregister {} student from tutorial group : {}", studentLogin, tutorialGroupId);
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdWithTeachingAssistantAndRegistrationsElseThrow(tutorialGroupId);
+        checkTutorialCourseIdMatchesPathId(courseId, tutorialGroupFromDatabase);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+
+        Optional<User> studentToDeregister = userRepository.findOneWithGroupsAndAuthoritiesByLogin(studentLogin);
+        if (studentToDeregister.isEmpty()) {
+            throw new EntityNotFoundException("User", studentLogin);
+        }
+        tutorialGroupService.deregisterStudent(studentToDeregister.get(), tutorialGroupFromDatabase);
+        return ResponseEntity.ok().body(null);
```

Comment:
Usually we return `ResponseEntity.noContent()` here.

Reply:
Done in 9eca40ec5100cab4110ea542c7bfc5de7a6f440e